### PR TITLE
feat: add option for `NewConsulRegisterWithConfig`

### LINKED ISSUE #8: Question: How to customize configuration while using Check
**Is your feature request related to a problem? Please describe.**

如何自定义 config 并且使用健康检查，`NewConsulRegisterWithConfig` 不能传 `Option` 我觉得不太好

### PR DESCRIPTION
**Pull request by L2ncE**
It seems cumbersome to use this middleware to configure tokens and use health checks at the same time. I think a new input parameter needs to be added.

Fixes #8

### CODE REVIEW

File: consul_registry.go [hunk 1]
```diff
@@ -62,13 +62,21 @@ func NewConsulRegister(address string, opts ...Option) (registry.Registry, error
 }
 
 // NewConsulRegisterWithConfig create a new registry using consul, with a custom config.
-func NewConsulRegisterWithConfig(config *api.Config) (*consulRegistry, error) {
+func NewConsulRegisterWithConfig(config *api.Config, opts ...Option) (*consulRegistry, error) {
```
Comment (baiyutang) [inline consul_registry.go:72]: Can you additional unit test and sample code ?

Comment (L2ncE) [reply consul_registry.go:72]: > Can you additional unit test and sample code ?

done

Comment (L2ncE) [reply consul_registry.go:65]: > Can you additional unit test and sample code ?

done

Comment (baiyutang) [reply consul_registry.go:72]: how to test this?
```
opts ...Option
```

File: example/custom-config/server/main.go [hunk 2]
```diff
@@ -20,13 +20,12 @@ import (
 	"context"
 	"log"
 
-	consulapi "github.com/hashicorp/consul/api"
-
+	"github.com/cloudwego/kitex-examples/hello/kitex_gen/api"
```
Comment (baiyutang) [inline example/custom-config/server/main.go]: Do not use this package， instead of `github.com/kitex-contrib/registry-consul/example/hello`

Comment (L2ncE) [reply example/custom-config/server/main.go]: done

**Review by baiyutang — APPROVED** [review]

**Pull request merged**